src/Microsoft.Azure.SignalR.Common/Generated/GeneratedRestClient.cs

```diff
@@ -0,0 +1,326 @@
+// <auto-generated>
```

Comment:
can auto-generated code be ignored?

Reply:
Currently the code is not generated during building. It has to be generated by Autorest which requireds Node.js environment. As the generated codes change little, I think it is more convenient to keep it here.

Reply:
So why not choose a library only requires dotnet?

What about adding a script to prepare the environment like this:
https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md

Reply:
Autorest is widely used in Azure-Powershell and azure-sdk-for-net projects and their common practice is just including the generated code into version control. I haven't found similar library like it only depending on dotnet, and if you know about this, we could have discussion. Actually I have tried to include the generation script into our build.cmd or build.sh and added Node.js environment into CI workflow but failed to start the build. Maybe I got something wrong with workflow files. 
I think another reason to upload the generated file is that we can review the generated code on Github.

Reply:
What about this one? https://github.com/RicoSuter/NSwag/wiki/CSharpClientGenerator

I think a new script is better, since build.sh/cmd is used for generating nuget package, we don't need to wait for the whole build.

Why we need to review the auto generated code, we just need to test the api whether the behavior is expected.

Reply:
There is already a script (client-generation.ps1)[https://github.com/Azure/azure-signalr/pull/992/files#diff-9c01492b97d08ccb869602c016adac19]